kubernetes/samples/scripts/domain-lifecycle/startServer.sh

```diff
@@ -0,0 +1,225 @@
+domainNamespace="sample-domain1-ns"
+keepReplicaConstant=false
+verboseMode=false
+withRelicas="CONSTANT"
+withPolicy="CONSTANT"
+managedServerPolicy=""
+action=""
+isValidServer=""
+patchJson=""
+serverStarted=""
+startsByPolicyUnset=""
+startsByReplicaIncreaseAndPolicyUnset=""
+
+while getopts "vkd:n:m:s:h" opt; do
+  case $opt in
+    s) serverName="${OPTARG}"
+    ;;
+    n) domainNamespace="${OPTARG}"
+    ;;
+    d) domainUid="${OPTARG}"
+    ;;
+    k) keepReplicaConstant=true;
+    ;;
+    m) kubernetesCli="${OPTARG}"
+    ;;
+    v) verboseMode=true;
+    ;;
+    h) usage 0
+    ;;
+    *) usage 1
+    ;;
+  esac
+done
+
+#
+# Function to perform validations, read files and initialize workspace
+#
+function initialize {
+
+  # Validate the required files exist
+  validateErrors=false
+
+  validateKubernetesCliAvailable
+  validateJqAvailable
+
+  # Validate that server name parameter is specified.
+  if [ -z "${serverName}" ]; then
+    validationError "Please specify a server name using '-s' parameter e.g. '-s managed-server1'."
+  fi
+
+  failIfValidationErrors
+}
+
+initialize
+
+# Get the domain in json format
+domainJson=$(${kubernetesCli} get domain ${domainUid} -n ${domainNamespace} -o json)
+if [ $? -ne 0 ]; then
+  echo "[ERROR} Unable to get domain resource. Please make sure 'domain_uid' and 'namespace' provided with '-d' and '-n' arguments are correct."
+  exit 1
+fi
+
+# Validate that specified server is either part of a cluster or is an independent managed server
+validateServerAndFindCluster "${domainUid}" "${domainNamespace}" "${serverName}" isValidServer clusterName
+if [ "${isValidServer}" != 'true' ]; then
+  printError "Server ${serverName} is not part of any cluster and it's not an independent managed server. Please make sure that server name specified is correct."
+  exit 1
+fi
+
+getClusterPolicy "${domainJson}" "${clusterName}" clusterPolicy
+if [ "${clusterPolicy}" == 'NEVER' ]; then
+  echo "The .spec.clusters[?(clusterName="${clusterName}"].serverStartPolicy of the domain resource is 'NEVER'. The $(basename $0) script will exit without starting server ${serverName}."
+  exit 0
+fi
+
+getDomainPolicy "${domainJson}" domainPolicy
+if [ "${domainPolicy}" == 'NEVER' ]; then
+  echo "The .spec.serverStartPolicy of the domain resource is 'NEVER'. The $(basename $0) script will exit without starting server ${serverName}."
+  exit 0
+fi
+
+getEffectivePolicy "${domainJson}" "${serverName}" "${clusterName}" effectivePolicy
+if [ -n "${clusterName}" ]; then
+  # Server is part of a cluster, check currently started servers
+  checkStartedServers "${domainJson}" "${serverName}" "${clusterName}" "${withRelicas}" "${withPolicy}" serverStarted
+  if [[ ${effectivePolicy} == "IF_NEEDED" && ${serverStarted} == "true" ]]; then
+    echo "[INFO] The server should be already started or it's starting. The start policy for server ${serverName} is ${effectivePolicy} and server is chosen to be started based on current replica count."
```

Comment:
Maybe let's have a zoom and word-smith the script's messages a little together?

Reply:
ok, sure. That would be helpful.